src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -454,6 +490,9 @@ public static bool DoesFragmentConditionMatch(ExecutionContext context, string f
             return false;
         }
 
+        /// <summary>
+        /// Returns a list of subfields (child nodes) for a result node based on the selection set from the document.
```

Comment:
I suggest to inline this method body into `ReadonlyResolveFieldContext.GetSubFields()` and remove it from here.

Reply:
I wouldn't.  `ReadonlyResolveFieldContext` happens to be the only use, but it shouldn't contain logic code.  And the code contained therein seems to be a good helper method.  If anyone wanted to make their own execution strategy along with their own `ReadonlyResolveFieldContext`, they should not have to re-implement `GetSubFields()`.

Reply:
All logic is inside `CollectFields`. `SubFieldsFor` is no more then a thin wrapper. And as you see there is only one client for this wrapper - `ReadonlyResolveFieldContext.GetSubFields()` so it seems to me that this method is in vain here.

Reply:
I don't care.  I wouldn't but it doesn't matter much.

Reply:
Of course it has to be done in the `develop` branch.

Reply:
OK. I will do it along with other changes.